elasticsearch/bulk/bulk.go

```diff
@@ -312,20 +314,34 @@ func (b *Bulk) CheckAndCommit() {
 func (b *Bulk) requestFunc(concurrentRequestIndex int, batchItems []BatchItem) func() error {
 	return func() error {
 		reader := b.readers[concurrentRequestIndex]
-		reader.Reset(getBytes(batchItems))
 		actionsOfBatchItems := getActions(batchItems)
-		r, err := b.esClient.Bulk(reader)
-		if err != nil {
-			b.finalizeProcess(actionsOfBatchItems, fillErrorDataWithBulkRequestError(actionsOfBatchItems, err))
-			return err
-		}
-		errorData, err := hasResponseError(r)
-		b.finalizeProcess(actionsOfBatchItems, errorData)
+		batchItemBytes := getBytes(batchItems)
+		reader.Reset(batchItemBytes)
+
+		for attempt := 1; attempt <= b.config.Elasticsearch.MaxRetries; attempt++ {
```

Comment:
Bu configi vermeyen clientlar, defaultta math.Maxınt oldugu için sonsuza kadar deneme ihtimali var mı? hep eof hep eof mesela return etmezse

Reply:
retrylar arasında delay olsa daha mi iyi olur?

Reply:
ve şu an sadece `io.ErrUnexpectedEOF` retry etmiş oluyoruz bu oki dmi

Reply:
Tamamdır eline sağlık kanka, complexleştirecekse gerek yok :D

Reply:
go-elasticsearch'de retry backoff u expose etmedigimiz icin bence de simdilik gerek yok. expose edersek buraya da ekleyebiliriz. :hype: